bot/view.py

```diff
@@ -0,0 +1,47 @@
+from abc import ABC, abstractmethod
+
+from fastapi import Request
+from fastapi.templating import Jinja2Templates
+
+from bot.controller import BotController
+
+
+class BotView(ABC):
+    @abstractmethod
+    def show_list_chatbots(self):
+        pass
+      
+    @abstractmethod
+    def show_create_chatbots(self):
+        pass
+    
+    @abstractmethod
+    def show_edit_chatbot(self, id: str):
+        pass
+    
+# note: authentication not impl yet
+class BotViewV1(BotView):    
+    def __init__(self, controller: BotController) -> None:
+        super().__init__()
+        self.templates = Jinja2Templates(directory="bot/templates")
+        self.controller = controller
+    
+    def show_list_chatbots(self, request: Request):
```

Comment:
code smell based on sonarcloud because the show_list_chatbots method in the abstract class BotView is defined without parameters, while the implementation in BotViewV1 includes a request: Request parameter.

Reply:
I see... can you show me the sonarcloud link / ref?

Reply:
nevermind, I see it. I think I missed the whole code smells section. thanks for raising the issue jason